common/djangoapps/third_party_auth/api/views.py

```diff
@@ -68,26 +99,22 @@ def get(self, request, username):
             JSON serialized list of the providers linked to this user.
 
         """
-        if request.user.username != username:
-            # We are querying permissions for a user other than the current user.
-            if not request.user.is_superuser and not ApiKeyHeaderPermission().has_permission(request, self):
-                # Return a 403 (Unauthorized) without validating 'username', so that we
-                # do not let users probe the existence of other user accounts.
+        is_unprivileged = self.is_unprivileged_query(request, username)
+        if is_unprivileged:
+            if not getattr(settings, 'ALLOW_UNPRIVILEGED_SSO_PROVIDER_QUERY', False):
                 return Response(status=status.HTTP_403_FORBIDDEN)
-
         try:
-            user = User.objects.get(username=username)
+            if '@' in username:
+                user = User.objects.get(email=username)
+            else:
+                user = User.objects.get(username=username)
```

Comment:
@pwnage101 Don't we have a method that does this?

Reply:
yes, use `from student.models import get_user_by_username_or_email`